packages/devextreme/js/ui/context_menu/ui.context_menu.js

```diff
@@ -556,6 +555,16 @@ class ContextMenu extends MenuBase {
         if(!arg.cancel) {
             this._hideAllShownSubmenus();
             this._setOptionWithoutOptionChange('visible', false);
+
+            const $scrollableElement = this._overlay.$content().find(`.${SCROLLABLE_CLASS}`).first();
+
+            if(!$scrollableElement.length) {
+                return;
+            }
+
+            const scrollableInstance = $scrollableElement.dxScrollable('instance');
+
+            scrollableInstance.scrollTo(0);
```

Comment:
[bad UX]
I see a flicking when i close root submenu. Cannot we call `scrollTo` later?

Reply:
I suggest u adding `onVisibilityChanged` handler for scrollable and pass it on scrollable init here. It will allow us:
- get rid of flicks
- leave the single place where we restore scroll position
- minimize count of imperative code